RA-03

kostyasmirnov opened the pull request:
В самом конце вспомнил что забыл сделать размер иконки на кнопке "Избранное". Добавил атрибут `app:iconSize="24dp"`, и кнопка избранное стало больше... 
Не смог понять как это исправит, прошу помочь

ievetrov commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -4,15 +4,46 @@
+        app:layout_constraintBottom_toBottomOf="parent">
+
+        <Button
+            android:id="@+id/button1"
```
Предлагаю давать более релевантные названия айдишникам, указывая, что это элемент кнопки и для чего он нужен. Кроме того индексы использовать не лучшая практика. Давай доработаем. Например, btnCategories

ievetrov commented on app/src/main/res/values/style.xml:
```diff
@@ -0,0 +1,20 @@
+    </style>
+
+    <style name="StyleNavButton" parent="Widget.Material3.Button">
+        <item name="android:textAppearance">@style/StyleTitle14</item>
```
смотри, этот стиль с регулар шрифтом перебивает то, что ниже прописал айтем с montserrat_alternates_medium. поэтому кнопки отображаются с тонким регулар шрифтом

ievetrov commented on app/src/main/res/values/style.xml:
```diff
@@ -0,0 +1,20 @@
+        <item name="android:fontFamily">@font/montserrat_alternates_medium</item>
+        <item name="android:paddingHorizontal">14dp</item>
+        <item name="android:paddingVertical">10dp</item>
+        <item name="android:textStyle">bold</item>
```
а этот элемент ни на что не влияет, тк вся необходимая толщина хранится в импортируемом шрифте

ievetrov commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -4,15 +4,46 @@
+            android:text="Избранное"
+            app:icon="@drawable/ic_favorites_heart"
+            app:iconGravity="textEnd"
+            app:iconSize="24dp" />
```
изначально ты импортировал иконку с нужным размером. как и в макете она такого же размера как текст. из-за повторного перемасштабирования картинка меняет свой размер.

ievetrov commented on app/src/main/res/values/themes.xml:
```diff
@@ -3,6 +3,9 @@
         <!-- <item name="colorPrimary">@color/my_light_primary</item> -->
+        <item name="android:statusBarColor">@color/main_activity_background_color</item>
+        <item name="android:windowLightStatusBar">true</item>
+        <item name="fontFamily">@font/montserrat_alternates_regular</item>
```
а чемуты решил такой шрифт задать для всего приложения?

kostyasmirnov commented:
fix: 
- added relevant names for ids
- changed the style for **StyleNavButton**
- delete `textStyle>bold<` in style **StyleNavButton**
- delete `StyleNavButton` in **btnFavorites**
- delete font from `themes.xml`

ievetrov commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -4,15 +4,45 @@
+            android:layout_weight="1"
+            android:backgroundTint="@color/favorites_btn_color"
+            android:drawablePadding="10dp"
+            android:text="Избранное"
```
Текст на кнопках не вынесен в текстовые ресурсы

ievetrov commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -4,15 +4,45 @@
-        app:layout_constraintEnd_toEndOf="parent"
-        app:layout_constraintStart_toStartOf="parent"
-        app:layout_constraintTop_toTopOf="parent" />
+        android:layout_marginStart="@dimen/main_padding"
```
Не обязательно для каждой стороны прописывать марджины по отдельности. Есть более универсальные атрибуты. 
android:layout_margin=""
android:layout_marginHorizontal=""
android:layout_marginVertical=""

ievetrov commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -4,15 +4,45 @@
-    <TextView
-        android:layout_width="wrap_content"
+    <LinearLayout
+        android:id="@+id/bottomButtons"
```
Предлагаю еще чуть более точно указать в айдишнике для чего он нужен, чтобы в коде потом проще было ориентироваться. Можно добавить в качестве префикса `layout...`

ievetrov commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -4,15 +4,45 @@
+        android:layout_marginEnd="@dimen/main_padding"
+        android:layout_marginBottom="@dimen/main_padding_half"
+        android:orientation="horizontal"
+        app:layout_constraintBottom_toBottomOf="parent">
```
Констрейнты корректнее задавать минимум на 2 оси. В данном случае рекомендую к твоему нижнему констрейнту добавить еще по бокам. От элемента к родителю.

ievetrov commented on app/src/main/res/values/style.xml:
```diff
@@ -0,0 +1,21 @@
+        <item name="android:textSize">14sp</item>
+        <item name="android:textAllCaps">true</item>
+        <item name="android:textColor">@color/text_btn_color</item>
+        <item name="fontFamily">@font/montserrat_alternates_regular</item>
```
А где еще используется regular шрифт? Ну то есть какое предназначение у него тут?

ievetrov commented on app/src/main/res/values/style.xml:
```diff
@@ -0,0 +1,21 @@
+    </style>
+
+    <style name="StyleNavButton" parent="Widget.Material3.Button">
+        <item name="android:textSize">14sp</item>
```
создание `StyleTitle14` было как раз для того, чтобы не дублировать тут значения толщины и размера шрифта. В названии стиля не случайно фигурирует `Title` - это значит, что таким шрифтом будут помечаться все заголовки. А они у нас преимущественно `SemiBold`.

Предлагаю из данного стиля кнопок убрать размер и толщину (которая medium и не соответствует макету). Затем отнаследоваться от StyleTitle14, где уже нужные есть толщина и размер шрифта.

kostyasmirnov commented:
fix: 
- texts on buttons are moved to text resources
- added universal attribute `android:layout_marginHorizontal="@dimen/main_padding"`
- renamed id layout
- added constraints left and right
- del font `montserrat_alternates_regular` from project
- del `textSize` and `fontFamily` from `StyleNavButton`
